src/main/java/com/redhat/devtools/lsp4ij/features/semanticTokens/viewProvider/LSPSemanticTokensFileViewProviderHelper.java

```diff
@@ -189,19 +191,62 @@ LSPSemanticToken getSemanticToken(int offset) {
         // If this file has semantic tokens, use them
         Map<Integer, LSPSemanticToken> semanticTokensByOffset = getSemanticTokensByOffset();
         if (!ContainerUtil.isEmpty(semanticTokensByOffset)) {
-            return semanticTokensByOffset.get(offset);
+            LSPSemanticToken semanticToken = semanticTokensByOffset.get(offset);
+            // Update the view provider's effective offset as appropriate
+            setEffectiveOffset(semanticToken == null ? offset : -1);
```

Comment:
If the requested offset corresponds to a semantic token PSI element, store that information.